### PR TITLE
CEM-1321 fileupload component

### PR DESCRIPTION
**Pull request by roggc**
https://drive.google.com/file/d/1IiAAxznLh5FHcGy2kxbG4shqD2bMA1h2/view?usp=sharing
disabled
dragenteranimation
disabledeffect

### CODE REVIEW

File: src/Containers/FileUpload/FileUpload.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { Meta, Story } from '@storybook/react';
+import {Image} from '@styled-icons/fa-solid/Image';
+import {TextLayout} from '@Layouts';
+import styled from 'styled-components'
+import { FileUpload,IFileUploadProps } from '../../index';
+import { createStoryTitle } from '../../Constants';
+
+const A=styled.a`
```
Comment (ralph-dev) [inline src/Containers/FileUpload/FileUpload.stories.tsx:9]: Bottom of the file

**Review by ralph-dev — COMMENTED** [review]
Lgtm, now onto event handlers

File: src/Containers/FileUpload/FileUpload.stories.tsx (hunk 1 above)
Comment (roggc) [reply src/Containers/FileUpload/FileUpload.stories.tsx:9]: no, because if not error about using a variable before declaring appears.

**Review by ralph-dev — COMMENTED** [review]
Looks really good, where is the success animation?

**Comment by ralph-dev** [comment]
Ok the component is looking better, but what about these cases 
1) When the file is successfully loaded does it appear anywhere?
2) If the Image Picker allows for only 1 file to be uploaded max how does the UI look for replacement.
3) If it allows more than one file how does it show a file that's been uploaded / allow you to upload more UI

File: src/Containers/FileUpload/FileUpload.tsx [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    isUploading,
+    isSuccess,
+    isFailure,
+    successMessage,
+    failureMessage,
+}):React.ReactElement=>{
+    const [height,setHeight]=useState<number|undefined>(undefined)
+    const [totalHeight,setTotalHeight]=useState(0)
+    const [totalHeightPlus,setTotalHeightPlus]=useState(0)
+    const [padding]=useState(10)
```
Comment (ralph-dev) [inline src/Containers/FileUpload/FileUpload.tsx:38]: Why is this here? If your using this for conditional animation it should be in styled components. If your using this for rendering an animations you should use CSS Keyframes

File: src/Containers/FileUpload/FileUpload.tsx [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+    const [maxHeight,setMaxHeight]=useState<number|undefined>(undefined)
+    const [positionLoading,setPositionLoading]=useState(!isUploading)
+    const [positionTopLoading,setPositionTopLoading]=useState(0)
+    const [opacityLoading,setOpacityLoading]=useState(0)
+    const [positionIsSuccess,setPositionIsSuccess]=useState(!isSuccess)
+    const [opacityIsSuccess,setOpacityIsSuccess]=useState(0)
+    const [loadingContainerHeight,setLoadingContainerHeight]=useState(0)
+    const [widthComponent,setWidthComponent]=useState<number>(0)
+    const [margin]=useState(10)
```
Comment (ralph-dev) [inline src/Containers/FileUpload/FileUpload.tsx:47]: No

File: src/Containers/FileUpload/FileUpload.tsx [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+    const [totalHeightPlus,setTotalHeightPlus]=useState(0)
+    const [padding]=useState(10)
+    const [maxHeight,setMaxHeight]=useState<number|undefined>(undefined)
+    const [positionLoading,setPositionLoading]=useState(!isUploading)
+    const [positionTopLoading,setPositionTopLoading]=useState(0)
+    const [opacityLoading,setOpacityLoading]=useState(0)
+    const [positionIsSuccess,setPositionIsSuccess]=useState(!isSuccess)
+    const [opacityIsSuccess,setOpacityIsSuccess]=useState(0)
+    const [loadingContainerHeight,setLoadingContainerHeight]=useState(0)
+    const [widthComponent,setWidthComponent]=useState<number>(0)
```
Comment (ralph-dev) [inline src/Containers/FileUpload/FileUpload.tsx:46]: Do you need to specify the number type?

File: src/Containers/FileUpload/FileUpload.tsx [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+            setPositionIsSuccess(true)
+        }
+        return ()=>{
+            setPositionIsSuccess(false)
+            setOpacityIsSuccess(1)
+        }
+    },[isSuccess,isFailure])
+
+
+    useEffect(()=>{
```
Comment (ralph-dev) [inline src/Containers/FileUpload/FileUpload.tsx:120]: Write descriptions for these use effects you have so many

File: src/Containers/FileUpload/FileUpload.tsx [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+            <Container withBorder padding='30px 20px 43px 20px' opacity={opacityLoading} position={positionLoading} ref={loadingContainerRef} overflow='hidden' width={widthComponent} margin={`${margin}px`} positionTop={positionTopLoading}>
+                <Loading loading={isUploading} message='Uploading...' />
+            </Container>
+            <Container withBorder opacity={opacityIsSuccess} height={loadingContainerHeight} position={positionIsSuccess} margin={`${margin}px`} positionTop={positionTopLoading} width={widthIsSuccess}>
+                <Container withFlexSpaceBetween>
+                    <TextLayout bold color='DarkBlue'>{successMessage}</TextLayout>
+                    <Icon as={CheckCircle} color={MainTheme.colors.statusColors.green} />
+                </Container>
+            </Container>
+            <Container withBorder opacity={opacityIsFailure} height={loadingContainerHeight} position={positionIsFailure} margin={`${margin}px`} positionTop={positionTopLoading} width={widthIsSuccess}>
```
Comment (ralph-dev) [inline src/Containers/FileUpload/FileUpload.tsx:193]: These should be sub components, becuase they are only used under this component

**Review by ralph-dev — CHANGES_REQUESTED** [review]
image

File: src/Containers/FileUpload/FileUpload.tsx (hunk 2 above)
Comment (roggc) [reply src/Containers/FileUpload/FileUpload.tsx:38]: I use this padding state variable to calculate (compute) the width of the element and set it when its position is absolute.

File: src/Containers/FileUpload/FileUpload.tsx (hunk 4 above)
Comment (roggc) [reply src/Containers/FileUpload/FileUpload.tsx:46]: no

File: src/Containers/FileUpload/FileUpload.tsx (hunk 2 above)
Comment (ralph-dev) [reply src/Containers/FileUpload/FileUpload.tsx:38]: I think this is incorrect, if you can explain more about why it's being used / how that will help suggest a better solution. Are you sure you don't need a useRef or anything else I'm sure there is a better way

File: src/Containers/FileUpload/BottomPanel.tsx [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import React,{forwardRef} from 'react'
+import {Container} from './Container'
+
+interface IBottomPanelProps{
```
Comment (ralph-dev) [inline src/Containers/FileUpload/BottomPanel.tsx:4]: Why is this not extending HTMLDivElemt so all the props don't need to be retyped??

File: src/Containers/FileUpload/Container.ts [hunk 8]
```diff
@@ -0,0 +1,43 @@
+import styled from 'styled-components'
+import {flex} from '@Utils/Mixins'
+
+interface IContainerProps{
```
Comment (ralph-dev) [inline src/Containers/FileUpload/Container.ts:4]: not extending correct interface

**Review by ralph-dev — CHANGES_REQUESTED** [review]
What is this code quality, please ask questions about TypeScript if you don't understand something.

File: src/Containers/FileUpload/BottomPanel.tsx [hunk 9]
```diff
@@ -0,0 +1,13 @@
+import React,{forwardRef} from 'react'
+import {Container,IContainerProps} from './Container'
+
+interface IBottomPanelProps extends IContainerProps{
+}
+
+export const BottomPanel=forwardRef<HTMLDivElement,IBottomPanelProps>(({children,...props},forwadedRef):React.ReactElement=>{
```
Comment (ralph-dev) [inline src/Containers/FileUpload/BottomPanel.tsx:7]: What's by is there a <> here

File: src/Containers/FileUpload/Container.ts [hunk 10]
```diff
@@ -0,0 +1,43 @@
+import styled from 'styled-components'
+import {flex} from '@Utils/Mixins'
+
+export interface IContainerProps{
```
Comment (ralph-dev) [inline src/Containers/FileUpload/Container.ts:4]: Why not extend Div?

**Review by ralph-dev — CHANGES_REQUESTED** [review]

File: src/Containers/FileUpload/BottomPanel.tsx (hunk 9 above)
Comment (roggc) [reply src/Containers/FileUpload/BottomPanel.tsx:7]: I don't understand sorry

File: src/Containers/FileUpload/Container.ts (hunk 10 above)
Comment (roggc) [reply src/Containers/FileUpload/Container.ts:4]: I think it's not necessary in the case of styled components.

File: src/Containers/FileUpload/BottomPanel.tsx (hunk 9 above)
Comment (ralph-dev) [reply src/Containers/FileUpload/BottomPanel.tsx:8]: Why do you have <Div, props>
Shouldn't the props just extend Div?

Comment (roggc) [reply src/Containers/FileUpload/BottomPanel.tsx:8]: This is how it is done when you need to forward a ref in typescript

File: src/Containers/FileUpload/Container.ts (hunk 10 above)
Comment (ralph-dev) [reply src/Containers/FileUpload/Container.ts:4]: But then why are you redeclaring are the styled props?

If you want all the style props exposed without doing this manually and having them optional why not just do

Partial<Pick<Div, Style>>

**Review by ralph-dev — CHANGES_REQUESTED** [review]
You component is missing the animation for onHover btw
I want you to integrate Lottie in this PR, it's very easy and will make your components higher quality, https://lottiefiles.com/search?q=Files&category=animations

File: src/Containers/FileUpload/BottomPanel.tsx (hunk 9 above)
Comment (ralph-dev) [reply src/Containers/FileUpload/BottomPanel.tsx:8]: But why do you need Div, Props
Can you just have Props
Then have props extend Div?

**Review by ralph-dev — CHANGES_REQUESTED** [review]
File Upload / Container File Interface what is that??
Everytime I find a mistake in the PR I am going to add an additional UI / UX requirement until your code is perfect 👍🏻 
New Requirement: Let's have the Dashboard Border Animates (As in Moves around like a train) so that the dashed lines are animating when you've picked a file up, like the animation :)

File: src/Containers/FileUpload/BottomPanel.tsx (hunk 9 above)
Comment (roggc) [reply src/Containers/FileUpload/BottomPanel.tsx:8]: HTMLDivElement refers to the ref type

File: src/Containers/FileUpload/Container.ts (hunk 10 above)
Comment (roggc) [reply src/Containers/FileUpload/Container.ts:4]: Sorry, I don't get it. What is Style? I have personal options (properties) defined by me which I need. That's why I declare them.

File: src/Containers/FileUpload/BottomPanel.tsx (hunk 9 above)
Comment (ralph-dev) [reply src/Containers/FileUpload/BottomPanel.tsx:8]: Then isn't the ordering incorrect?
Becuase the ref is the second parameter not the first

File: src/Containers/FileUpload/Container.ts (hunk 10 above)
Comment (ralph-dev) [reply src/Containers/FileUpload/Container.ts:4]: I do not understand what is happening here at all. Can you try to explain what EXACTLY you are trying to do here.
DragEnter comes from https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.dragevent.html 
DragEvent, i don't know why you are creating custom types? Margin is already from div elements etc

File: src/Containers/FileUpload/FileUpload.tsx [hunk 11]
```diff
@@ -0,0 +1,236 @@
...
+        return ()=>{
+            setPositionLoading(false)
+            setOpacityLoading(1)
+        }
+    },[isUploading,isSuccess,isFailure])
+
+    // this is to fade out success container component
+    useEffect(()=>{
+        if(!isSuccess||isFailure){
+            setOpacityIsSuccess(0)
```
Comment (ralph-dev) [inline src/Containers/FileUpload/FileUpload.tsx:120]: Why is opacity have loading / success states? Can the naming be improved

File: src/Containers/FileUpload/FileUpload.tsx [hunk 12]
```diff
@@ -0,0 +1,236 @@
...
+                <Container withFlexSpaceBetween>
+                    <TextLayout bold color='DarkBlue'>{successMessage}</TextLayout>
+                    <Icon as={CheckCircle} color={MainTheme.colors.statusColors.green} />
+                </Container>
+            </BottomPanel>
+            <BottomPanel withBorder opacity={opacityIsFailure} height={loadingContainerHeight} position={positionIsFailure} margin={`${margin}px`} positionTop={positionTopLoading} width={widthIsSuccess}>
+                <Container withFlexSpaceBetween>
+                    <TextLayout bold color='DarkBlue'>{failureMessage}</TextLayout>
+                    <Icon as={TimesCircle} color={MainTheme.colors.statusColors.red} />
+                </Container>
```
Comment (ralph-dev) [inline src/Containers/FileUpload/FileUpload.tsx:207]: Why aren't you re-using code????

**Review by ralph-dev — CHANGES_REQUESTED** [review]
Getting better still work left to do

File: src/Containers/FileUpload/FileUpload.tsx (hunk 11 above)
Comment (roggc) [reply src/Containers/FileUpload/FileUpload.tsx:120]: because there is opacity for the isUploading panel, opacity for the isSuccess panel and opacity for the isFailure panel

File: src/Containers/FileUpload/FileUpload.tsx (hunk 12 above)
Comment (roggc) [reply src/Containers/FileUpload/FileUpload.tsx:207]: I don't know how I can reuse code in this case. BottomPanel renders it's children, which in this case there are three different bottom panel with three different children (there are two similar, but one different, which is the isUploading bottompanel)

Comment (ralph-dev) [reply src/Containers/FileUpload/FileUpload.tsx:207]: Children as a prop then?

Comment (ralph-dev) [reply src/Containers/FileUpload/FileUpload.tsx:207]: Just inject the things that are different as props

**Comment by ralph-dev** [comment]
New requirement: When in Disbaled State, UI should have opacity faded or look disabled!

File: src/Containers/FileUpload/FileUpload.tsx (hunk 12 above)
Comment (roggc) [reply src/Containers/FileUpload/FileUpload.tsx:207]: Yes, that's it. children it's already a prop. The only thing that occurs to me is to make a react component for the two similar children.

**Comment by ralph-dev** [comment]
@roggc Which gif contains this case?
Max One File, With One File Already Uploaded (Thus Disabled becuase max files reached?)
Also fix merge conflicts

**Comment by roggc** [comment]
@ralph-dev , the last gif where the disabled property is turned on should be the case where no more files are allowed.

File: src/Containers/FileUpload/BottomPanel.tsx (hunk 9 above)
Comment (roggc) [reply src/Containers/FileUpload/BottomPanel.tsx:8]: Yes, that's right, but this is how it is. blame react! hehe

File: src/Containers/FileUpload/Container.ts (hunk 10 above)
Comment (roggc) [reply src/Containers/FileUpload/Container.ts]: Is not DrageEnter, is isDragEnter, which is a boolean. and margin is not a valid div attribute, as far as I know

**Review by Ampatte2 — COMMENTED** [review]
I see a lot of useState and useEffect. This code is difficult to understand and read, I am thinking this may be better with a reducer for a few reasons. 

Clarity of the naming/variables. Instead of 10 different states now you can organize variables within your single state such as creating properties like loading { opacity: 0, position: 0}, isSuccess {position: 0, opacity: 0}. 

Instead of multiple useEffects to update you can have one useEffect trigger an action based on conditions. And that action is named to give the reader an idea of what those conditions mean together.

For something as complicated as this component I think being verbose is definitely the way to go, if I needed to debug this in the current state I would need to learn the entire thing.

File: src/Containers/FileUpload/FileUpload.tsx [hunk 13]
```diff
@@ -38,135 +188,135 @@ export const FileUpload:React.FC<IFileUploadProps>=({
...
+    },[isUploading,isSuccess,isFailure,state.totalHeight,state.totalHeightPlus])
 
     // this is to calculate (set) some values
     useEffect(()=>{
         if(containerRef.current?.scrollHeight){
             const contentHeight=containerRef.current.scrollHeight
-            setTotalHeight(contentHeight-padding*2)
-            setMaxHeight(contentHeight-padding*2)
+            dispatch({type:SET_TOTALHEIGHT,value:contentHeight-state.padding*2})
+            dispatch({type:SET_MAXHEIGHT,value:contentHeight-state.padding*2})
```
Comment (Ampatte2) [inline src/Containers/FileUpload/FileUpload.tsx:228]: These can be combined into one dispatch, and the same for line 215.

File: src/Containers/FileUpload/FileUpload.tsx [hunk 14]
```diff
@@ -38,135 +188,135 @@ export const FileUpload:React.FC<IFileUploadProps>=({
...
         }
         if(rootRef.current?.clientWidth){
-            setWidthComponent(rootRef.current?.clientWidth-margin*2-padding*2)
+            dispatch({type:SET_COMPONENTWIDTH,value:rootRef.current.clientWidth-state.margin*2-state.padding*2})
```
Comment (Ampatte2) [inline src/Containers/FileUpload/FileUpload.tsx:231]: I am not the biggest fan of computing values within setting a state or dispatching, declaring your computed value with a named variable then passing that into the dispatch/state gives more information about what is happening.

File: src/Containers/FileUpload/FileUpload.tsx [hunk 15]
```diff
@@ -38,135 +188,135 @@ export const FileUpload:React.FC<IFileUploadProps>=({
...
-    },[isSuccess,isFailure])
+    },[isSuccess,isFailure,state.padding,state.margin])
 
     // this is to fade out uploading container component
     useEffect(()=>{
         if(!isUploading||isSuccess||isFailure){
-            setOpacityLoading(0)
-            setPositionLoading(true)
+            dispatch({type:SET_OPACITYLOADING,value:0})
+            dispatch({type:SET_POSITIONLOADING,value:true})
```
Comment (Ampatte2) [inline src/Containers/FileUpload/FileUpload.tsx:261]: Combine and name.

**Review by Ampatte2 — COMMENTED** [review]
This mainly converts states into actions, which doesn't solve the issue that it is complicated. There are some dispatches which are called together and can be combined.

There are five useEffects that have similiar dependancies of isSuccess, isFailure, and isUploading (in various combinations), is it possible to combine these and handle certain actions based on common conditions? (Let me know what you think).

After combining the actions is there a way to better name the them to tell the reader what change this is effecting or why this action is changing the parts of the state it is?

File: src/Containers/FileUpload/FileUpload.tsx (hunk 13 above)
Comment (roggc) [reply src/Containers/FileUpload/FileUpload.tsx:228]: how do you combine it into only one dispatch?

File: src/Containers/FileUpload/FileUpload.tsx (hunk 15 above)
Comment (roggc) [reply src/Containers/FileUpload/FileUpload.tsx:261]: how do you combine?

Comment (roggc) [reply src/Containers/FileUpload/FileUpload.tsx:261]: ok, I know it. Let me try it...

File: src/Containers/FileUpload/FileUpload.tsx (hunk 13 above)
Comment (roggc) [reply src/Containers/FileUpload/FileUpload.tsx:228]: yes, I know it. I will do it...

File: src/Containers/FileUpload/FileUpload.tsx [hunk 16]
```diff
@@ -157,6 +165,60 @@ const reducer=(state:IState,action:Action):IState=>{
...
+        }
+    case ISFAILURE_RESTORE:
+        return {
+            ...state,
+            isFailure:{
+                ...state.isFailure,
+                opacity:1,
+                position:false,
+            },
+        }
```
Comment (Ampatte2) [inline src/Containers/FileUpload/FileUpload.tsx:221]: Yes this is much better! I have a better understanding of why this action is taking place!

**Review by Ampatte2 — COMMENTED** [review]
There are other dispatches like line 277 and 289 that can be combined. Still needs some actions to be renamed to be more descriptive. The convention of ISSUCCESS_FADEOUT would be a boolean, SUCCESS_FADEOUT is sufficient. 

Let's try to condense the actions more and combine useEffects that are dependant on isSuccess, isFailure, and isUploading.

File: src/Containers/FileUpload/FileUpload.tsx [hunk 17]
```diff
@@ -350,7 +350,8 @@ export const FileUpload:React.FC<IFileUploadProps>=({
     useLayoutEffect(() => {
         function updateSize() {
             if(rootRef.current?.clientWidth){
-                dispatch({type:SET_COMPONENTWIDTH,value:rootRef.current.clientWidth-state.margin*2-state.padding*2})
+                const width=rootRef.current.clientWidth-state.margin*2-state.padding*2
```
Comment (Ampatte2) [inline src/Containers/FileUpload/FileUpload.tsx:353]: Needs better naming, what is this calculation performing? Descriptive is better than short.

File: src/Containers/FileUpload/BottomPanel.tsx (hunk 9 above)
Comment (ralph-dev) [reply src/Containers/FileUpload/BottomPanel.tsx:8]: Oh my god you are right haha this is so inconsistent with everything else, god damn React!!

File: src/Containers/FileUpload/FileUpload.tsx [hunk 18]
```diff
@@ -0,0 +1,451 @@
...
+        return {
+            ...state,
+            totalHeightPlus:action.value
+        }
+    case SET_POSITIONTOPLOADING:
+        return {
+            ...state,
+            positionTopLoading:action.value
+        }
+    case SET_ISSUCCESSWIDTH:
```
Comment (ralph-dev) [inline src/Containers/FileUpload/FileUpload.tsx:107]: SET_IS_SUCCESS_WIDTH

SCREAMING_SNAKE_CASE

File: src/Containers/FileUpload/FileUpload.tsx [hunk 19]
```diff
@@ -0,0 +1,456 @@
...
+    isSuccess:IOptions;
+    isFailure:IOptions;
+    positionTopLoading:number;
+    loadingContainerHeight:number;
+    componentWidth:number;
+    isSuccessWidth:number;
+    isDragEnter:boolean;
+}
+
+const SET_HEIGHT='SET_HEIGHT'
```
Comment (ralph-dev) [inline src/Containers/FileUpload/FileUpload.tsx:38]: What is this?

**Review by ralph-dev — CHANGES_REQUESTED** [review]
?

File: src/Containers/FileUpload/FileUpload.tsx (hunk 19 above)
Comment (roggc) [reply src/Containers/FileUpload/FileUpload.tsx:38]: this is the redux way to declare types of actions

**Review by ralph-dev — APPROVED** [review]
LGTM, will let Andrew approve as well

File: src/Containers/FileUpload/FileUpload.tsx [hunk 20]
```diff
@@ -421,15 +432,9 @@ export const FileUpload:React.FC<IFileUploadProps>=({
...
-                <Loading loading={isUploading} message='Uploading...' />
-            </BottomPanel>
-            <BottomPanel withBorder opacity={state.isSuccess.opacity} height={state.loadingContainerHeight} position={state.isSuccess.position} margin={`${state.margin}px`} positionTop={state.positionTopLoading} width={state.isSuccessWidth}>
-                <IsFailureIsSuccessPanel message={successMessage} iconColor={MainTheme.colors.statusColors.green} IconToShow={CheckCircle} />
+            <BottomPanel withBorder padding={isFailure||isSuccess?undefined:'30px 20px 43px 20px'} opacity={state.loading.opacity||state.isSuccess.opacity||state.isFailure.opacity} position={state.loading.position&&state.isFailure.position&&state.isSuccess.position} ref={loadingContainerRef} overflow='hidden' width={isFailure||isSuccess?state.isSuccessWidth:state.componentWidth} margin={`${state.margin}px`} positionTop={state.positionTopLoading} height={isFailure||isSuccess?state.loadingContainerHeight:undefined}>
+                {renderChild()}
```
Comment (Ampatte2) [inline src/Containers/FileUpload/FileUpload.tsx:436]: This no longer makes sense to have all these opacities here. As we talked about yesterday this Bottompanel is a container for the children rendered inside of it. Create this panel and transition the opacity and height, it is either displayed or not.

Now, opacity needs to be added to the rendered components. I think this can be achieved by adding a @keyframes to the IsFailureIsSuccessPanel for opacity from 0 to 1 with 0.5s animation. There may be another way to do this.

Padding should also be static, I don't see why it would need to resize.

Add appropriate CSS to BottomPanel/IsFailureIsSuccess so that elements are not at edge of div.

**Review by Ampatte2 — CHANGES_REQUESTED** [review]

**Comment by Ampatte2** [comment]
Also padding and margin from the state should be static variables as they do not change.

**Review by ralph-dev — APPROVED** [review]
Still room for improvements but otherwise a LGTM

File: src/Containers/FileUpload/FileUpload.tsx [hunk 21]
```diff
@@ -35,6 +33,9 @@ interface IState{
     isDragEnter:boolean;
 }
 
+const padding=10;
+const margin=10;
```
Comment (Ampatte2) [inline src/Containers/FileUpload/FileUpload.tsx:37]: PADDING
MARGIN
declare outside component.

**Review by Ampatte2 — APPROVED** [review]
Small change and then LGTM

File: src/Containers/FileUpload/FileUpload.tsx (hunk 21 above)
Comment (roggc) [reply src/Containers/FileUpload/FileUpload.tsx:37]: just one thing, in the javascript style guide of the docs it says camelCase is permitted when declaring deeply immutable constants.

Comment (Ampatte2) [reply src/Containers/FileUpload/FileUpload.tsx:37]: https://wolfbeacon.atlassian.net/wiki/spaces/CW/pages/1130102818/JavaScript+Style+Guide#2.4.-Use-Upper-Case-for-Deeply-Immutable-Constants

I would go with the good option rather than permitted if I have a choice?

Comment (roggc) [reply src/Containers/FileUpload/FileUpload.tsx]: in fact in the docs says permitted for uppercase and not required to use uppercase. I think it's a matter of taste
